sdk/test-utils/recorder/src/utils/requestBodyTransform.ts

```diff
@@ -0,0 +1,50 @@
+export type RequestBodyType = string | { [x: string]: unknown };
+/**
+ * Under construction
+ */
+export function applyRequestBodyTransformations(
+  runtime: "node" | "browser",
+  fixture: string | { [x: string]: unknown },
+  requestBodyTransformations: (
+    | ((body: string) => string)
+    | ((body: { [x: string]: unknown }) => { [x: string]: unknown })
+  )[]
+): string | { [x: string]: unknown } {
+  if (runtime === "node") {
+    if (typeof fixture !== "string") {
+      return fixture;
+    }
+    // Modify the request body
+    let updatedFixture = fixture;
+    // console.log(`updatedFixture(stage 0) ==> ${updatedFixture}`);
+    // PUT and PATCH may also have request bodies, currently focusing only on POST - can be extended as needed
+    let matches = fixture.match(/\.post\((.*)\, (.*)\)\n\s*.reply\(/);
+    // console.log(matches);
+    if (matches?.[2] && typeof matches[2] === "string") {
+      let updatedBody = matches[2]; // Must be string - either normal or JSON-stringified
+      // normal string
+      for (const transformation of requestBodyTransformations) {
+        updatedBody = transformation(updatedBody as any) as string;
+      }
+      updatedFixture = fixture.replace(matches[2], updatedBody);
+      // TODO: JSON stringified
+    }
+    // console.log(`updatedFixture(stage 1) ==> ${updatedFixture}`);
+    // Modify the fixture with .filteringRequestBody method
+
+    matches = updatedFixture.match(/\.post\((.*)\, (.*)\)\n\s*.reply\(/);
+    if (matches?.[0]) {
+      for (const transformation of requestBodyTransformations) {
+        console.log(`.filteringRequestBody(${transformation.toString()})`);
+        updatedFixture = updatedFixture.replace(
+          matches[0],
+          `.filteringRequestBody(${transformation.toString()})\n  ` + matches[0]
```

Comment:
I don't like this transformation of the recording. Though it works, it may cause problems if nock ever decides to break things.

I'm planning to work on migrating to recorder v2 with JSON recordings for both node and browser.

Reply:
@jeremymeng @sadasant